### PR TITLE
Molecule role

### PR DESCRIPTION
**Pull request by jeffcpullen**
Merging in working working example of Molecule being run within Dev Spaces using 'oc exec' to interact with the pods without any modifications required to OpenShift.

### CODE REVIEW

File: demo/oc_connection/molecule/default/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+lint: |
+  set -e
+  yamllint .
+  ansible-lint .
+scenario:
+  name: default
+  test_sequence:
+    # - lint
+    # - syntax
+    # - prepare
```
Comment (kyetter) [inline demo/oc_connection/molecule/default/molecule.yml]: Do you want to list prepare as one of the sequence steps?

File: demo/oc_connection/molecule/default/prepare.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        molecule_user_home_dir: "{{ lookup('ansible.builtin.env', 'HOME') }}"
+        molecule_user_path: "{{ lookup('ansible.builtin.env', 'PATH') }}"
+
+    - name: Create local bin directory
+      ansible.builtin.file:
+        path: "{{ molecule_user_home_dir }}/bin"
+        state: directory
+        mode: 0770
+    
+    - name: Download foo.conf
```
Comment (kyetter) [inline demo/oc_connection/molecule/default/prepare.yml]: Need better name

File: demo/oc_connection/molecule/default/prepare.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        dest: "{{ molecule_user_home_dir }}/bin/oc"
+        mode: '0770'
+    
+    - name: Add another bin dir to system-wide $PATH.
+      ansible.builtin.lineinfile:
+        path: "{{ molecule_user_home_dir }}/.bashrc"
+        line: 'PATH=$PATH:{{ molecule_user_home_dir }}/bin'
+        state: present
+
+    # - name: Install oc client
```
Comment (kyetter) [inline demo/oc_connection/molecule/default/prepare.yml]: remove unless you're saving for something?

File: demo/oc_connection/molecule/default/templates/test.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "test1"
+  namespace: "jeffcpullen-devspaces"
```
Comment (kyetter) [inline demo/oc_connection/molecule/default/templates/test.yml]: Is there a way to not link this to your user? / peeking around I don't see this file being used? I might be missing it (?)

File: demo/oc_connection/requirements.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+collections:
+  - name: kubernetes.core
+    version: 2.4.0
+  - name: community.okd
+    version: 2.3.0
+  # Requires Hub / RH Token configured
```
Comment (kyetter) [inline demo/oc_connection/requirements.yml]: If we don't need lets clean up!

**Review by kyetter — COMMENTED** [review]
Small clean up notes, but yay yay yay!!!!

I played around with trying to automate the source bashrc and didn't have any luck - I'm good to capture the technical debt in the read me

File: demo/oc_connection/requirements.yml (hunk 5 above)
Comment (jeffcpullen) [reply demo/oc_connection/requirements.yml]: Fixed

File: demo/oc_connection/molecule/default/templates/test.yml (hunk 4 above)
Comment (jeffcpullen) [reply demo/oc_connection/molecule/default/templates/test.yml]: extra file, don't need. removed

File: demo/oc_connection/molecule/default/prepare.yml (hunk 3 above)
Comment (jeffcpullen) [reply demo/oc_connection/molecule/default/prepare.yml]: Removed

File: demo/oc_connection/molecule/default/prepare.yml (hunk 2 above)
Comment (jeffcpullen) [reply demo/oc_connection/molecule/default/prepare.yml]: Updated name, good catch

File: demo/oc_connection/molecule/default/molecule.yml (hunk 1 above)
Comment (jeffcpullen) [reply demo/oc_connection/molecule/default/molecule.yml]: Uncommented for the full suite

**Pull request merged**